include/rosidl_dynamic_typesupport/dynamic_message_type_support_struct.h

```diff
@@ -89,21 +90,53 @@ typedef struct rosidl_dynamic_message_type_support_impl_s
  */
 ROSIDL_DYNAMIC_TYPESUPPORT_PUBLIC
 rcutils_ret_t
-rosidl_dynamic_message_type_support_handle_create(
+rosidl_dynamic_message_type_support_handle_init(
   rosidl_dynamic_typesupport_serialization_support_t * serialization_support,
   const rosidl_type_hash_t * type_hash,
   const rosidl_runtime_c__type_description__TypeDescription * type_description,
   const rosidl_runtime_c__type_description__TypeSource__Sequence * type_description_sources,
-  rosidl_message_type_support_t ** ts);  // OUT
+  rcutils_allocator_t * allocator,
+  rosidl_message_type_support_t * ts);  // OUT
 
-/// Destroy a rosidl_message_type_support_t obtained with
-/// `rosidl_dynamic_message_type_support_handle_create()`, which has dynamically allocated members
+/// Finalize a rosidl_message_type_support_t obtained with
+/// `rosidl_dynamic_message_type_support_handle_init()`, which has dynamically allocated members
 ///
 /// NOTE: Using this on a statically allocated typesupport will cause undefined behavior!
 ///       (Static memory will get freed in that case.)
 ROSIDL_DYNAMIC_TYPESUPPORT_PUBLIC
 rcutils_ret_t
-rosidl_dynamic_message_type_support_handle_destroy(rosidl_message_type_support_t * ts);
+rosidl_dynamic_message_type_support_handle_fini(rosidl_message_type_support_t * ts);
+
+/// Initialized a `rosidl_dynamic_message_type_support_impl_t` with bound message description
+/**
+ * The `type_hash`, `type_description`, `type_description_sources`, and `allocator` arguments are
+ * copied.
+ *
+ * The caller is responsible for deallocating the returned pointer.
```

Comment:
Again, not sure what you're talking about here. Do you mean the `ts_impl`? If so, I think it's perhaps obvious that this is the case. They have to create something to pass in, so they have to clean it up. It might be that they pass in a reference to a thing created on the stack, in which case there's no deallocating. Do you instead mean that they need to call fini on it?

Reply:
Right, the context has changed, I need to update the docstrings..